src/licensedcode/models.py

```diff
@@ -1950,6 +2008,18 @@ def _from_expression(cls, license_expression=None, identifier=None, **kwargs):
         rule.setup()
         return rule
 
+    def compute_unique_id(self):
+        """
+        Return a a unique id string based on this rule content. (Today this is
+        an MD5 checksum of the text, but that's an implementation detail)
+        """
+        if not self.text:
+            text = "None"
```

Comment:
We cannot have all the items without a text having the same id. You need to inject something else then IMHO, unless you are sure this will not be an issue. But then a UUID may be better?

Reply:
I updated the code to be:
```
        content = f'{self.identifier!r}{self.license_expression!r}{self.text!r}'
        return sha1(content.encode('utf-8')).hexdigest()
```